primitives/utils/src/metrics.rs

```diff
@@ -0,0 +1,147 @@
+
+use lazy_static::lazy_static;
+use prometheus::{
+    Registry, Error as PrometheusError, Opts,
+    core::{
+        GenericGaugeVec, AtomicU64,
+        GenericGauge,
+	}
+};
+use std::time::Instant;
+use std::convert::TryFrom;
+use parking_lot::Mutex;
+
+use crate::channels::{ Sender, Receiver, OnFullStrategy, channel_with_strategy };
+
+lazy_static! {
+
+    pub static ref TOKIO_THREADS_TOTAL: GenericGauge<AtomicU64> = GenericGauge::new(
+        "tokio_threads_total", "Total number of threads created"
+    ).expect("Creating of statics doesn't fail. qed");
+
+    pub static ref TOKIO_THREADS_ALIVE: GenericGauge<AtomicU64> = GenericGauge::new(
+        "tokio_threads_alive", "Number of threads alive right now"
+    ).expect("Creating of statics doesn't fail. qed");
+
+	pub static ref UNBOUNDED_CHANNELS_COUNTER : GenericGaugeVec<AtomicU64> = GenericGaugeVec::new(
+        Opts::new("internals_unbounded_channels", "items in each mpsc::unbounded instance"),
```

Comment:
```suggestion
        Opts::new("unbounded_channel_len", "items in each mpsc::unbounded instance"),
```

I would find this metric name more descriptive.